### PR TITLE
Create new reminder #7

### LINKED ISSUE #7: Create A New Reminder
At the bottom of the `reminders.hbs` template, there should be a button that routes the user to `'/new'` where users can create a new reminder.

* The new reminder should have inputs for the title, date, and notes.
* The new reminder should have a button for submitting the new note.
* Pressing the button, should submit reminder to the database and have it shown in the Reminders list.

**Hot Tip**: TDD is really helpful.

### PR DESCRIPTION
**Pull request by ccgamble**
@martensonbj @stevekinney @nfosterky @Tman22 @nnchambs 

## Purpose

The purpose of this issue was to add a reminder to the list of reminders through input fields.

## Approach

First we created a button on the reminders page to 'Add a reminder'. Once this is clicked, the path is redirected to '/new' and input fields for adding a new reminder are displayed. When the user clicks the submit button, they are added to the list of reminders and rendered on the page.

### Learning

We consulted the Testing Ember lesson and Ember-groceries repo for this issue.

### Open Questions and Pre-Merge TODOs

- [x] There should be a button that routes the user to '/new' where users can create a new reminder
- [x]  The new reminder should have inputs for the title, date, and notes.
- [x] The new reminder should have a button for submitting the new note.
- [x] Pressing the button, should submit reminder to the database and have it shown in the Reminders list.
- [x] There should be acceptance tests for what has been added or changed.

### Test coverage 

To test that this new functionality was working properly, we created a test to make sure when the `Add new reminder` button is clicked, the path is changed to the correct route. Next, we tested that when the user is at the '/new' route, three input fields (for title, date, and notes) renders.  Last, we created a test to ensure that on submit this reminder is rendered to the reminders list with the correct values.

### Merge Dependencies and Related Work

This closes #7.

### CODE REVIEW

File: app/components/new-reminder.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  classNames: ['add-reminder'],
+
+  title: '',
+  date: '',
+  notes: '',
+
+  actions: {
+    createReminder() {
+      const reminder = this.getProperties('title', 'date', 'notes');
+			reminder.date = new Date(reminder.date);
```
Comment (nfosterky) [inline app/components/new-reminder.js]: Fix your indentation

File: app/components/new-reminder.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  store: Ember.inject.service(),
+
+  tagName: 'section',
```
Comment (nfosterky) [inline app/components/new-reminder.js]: 👍

File: app/router.js [hunk 3]
```diff
@@ -9,6 +10,7 @@ const Router = Ember.Router.extend({
 Router.map(function() {
   this.route('reminders', function() {
     this.route('reminder', {path: ':reminder_id'});
+		this.route('new', {path: '/new'});
```
Comment (nfosterky) [inline app/router.js]: 1. Fix indentation
2. You don't need to specify a path of /new as this is default for the 'new' route

File: app/templates/reminders.hbs [hunk 4]
```diff
@@ -1,10 +1,21 @@
...
 		<p>{{reminder.date}}</p>
 	</li>
 	{{/each}}
 </ul>
 
+
+
+
+<button >
+{{#link-to 'reminders.new' class="add-btn"}}Add New Reminder{{/link-to}}</button>
```
Comment (nfosterky) [inline app/templates/reminders.hbs]: fix indentation and move `</button>` to a new line

File: app/templates/reminders.hbs [hunk 5]
```diff
@@ -1,10 +1,21 @@
+<div>
+	<h2>Reminders</h2>
+
 	<ul>
 	{{#each model as |reminder|}}
-	<li>
+	<li class='reminder'>
 		<h3>{{#link-to 'reminders.reminder' reminder.id class="spec-reminder-item"}} {{reminder.title}} {{/link-to}}</h3>
```
Comment (nfosterky) [inline app/templates/reminders.hbs]: Seperate this into multiple lines to make it easier to read

**Comment by ccgamble** [comment]
@nfosterky I fixed the requested changes and pushed them back up to this branch.

File: app/templates/reminders.hbs [hunk 6]
```diff
@@ -2,19 +2,21 @@
...
-		<h3>{{#link-to 'reminders.reminder' reminder.id class="spec-reminder-item"}} {{reminder.title}} {{/link-to}}</h3>
-		<p>{{reminder.date}}</p>
-	</li>
-	{{/each}}
-</ul>
+		{{#each model as |reminder|}}
+			<li class='reminder'>
+				<h3>
+					{{#link-to 'reminders.reminder' reminder.id class="spec-reminder-item"}}
+					{{reminder.title}}
```
Comment (nfosterky) [inline app/templates/reminders.hbs]: Indent

File: mirage/scenarios/default.js [hunk 7]
```diff
@@ -1,3 +1,3 @@
-export default function(server) {
-  server.createList('reminder', 5);
+export default function() {
+  // server.createList('reminder', 5);
```
Comment (martensonbj) [inline mirage/scenarios/default.js]: Is there a reason to keep this file?

**Pull request merged**